### PR TITLE
[react-select] fix field name in AutocompleteResult

### PR DESCRIPTION
**Pull request by vmaark**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [ ] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by vujevits** [comment]
@Hesquibet @giladgray what do you think?

**Comment by skirsdeda** [comment]
👍 
Yup, seems correct.

File: react-select/react-select-tests.tsx [hunk 1]
```diff
@@ -158,9 +158,12 @@ class SelectWithStringValueTest extends React.Component<React.Props<{}>, {}> {
 class SelectAsyncTest extends React.Component<React.Props<{}>, {}> {
 
     render() {
-        const getOptions = (input: string, callback: Function) => {
+        const getOptions = (input: string, callback: (err: any, result: AutocompleteResult) => any) => {
```
Comment (giladgray) [inline react-select/react-select-tests.tsx]: very minor: `=> void` as the function never returns anything.

**Comment by giladgray** [comment]
👍 LGTM

**Comment by vujevits** [comment]
Can someone please merge this pr?

**Pull request merged**